code/modules/mob/living/simple_animal/hostile/ordeal/gold.dm

```diff
@@ -523,6 +523,8 @@
 	playsound(src, 'sound/effects/ordeals/gold/pridespin.ogg', 125, FALSE)
 
 /mob/living/simple_animal/hostile/ordeal/sin_pride/proc/Charge(move_dir, times_ran)
+	if(health < 0)
```

Comment:
Health is never below 0. Check stat and if you are really paranoid: `if(health <= 0)`